components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/AdditionalSubscriptionInfoMappingUtil.java

```diff
@@ -0,0 +1,467 @@
+                                                endpointUrls.add(solaceURLsDTO);
+                                            }
+                                            solaceDeployedEnvironmentsDTO.setSolaceURLs(endpointUrls);
+                                        }
+                                        // Get topic permissions from the solace application response body
+                                        if (environmentObject.getJSONObject("permissions") != null) {
+                                            org.json.JSONObject permissionsObject = environmentObject.getJSONObject
+                                                    ("permissions");
+                                            AdditionalSubscriptionInfoSolaceTopicsObjectDTO solaceTopicsObjectDTO =
+                                                    new AdditionalSubscriptionInfoSolaceTopicsObjectDTO();
+                                            populateSolaceTopics(solaceTopicsObjectDTO, permissionsObject,
+                                                    "default");
+                                            // Handle the special case of MQTT protocol
+                                            if (containsMQTTProtocol) {
+                                                HttpResponse responseForMqtt = solaceAdminApis.applicationGet
+                                                        (additionalSubscriptionInfoDTO.getSolaceOrganization(),
+                                                                application.getUUID(), SolaceConstants.
+                                                                        MQTT_TRANSPORT_PROTOCOL_NAME.toUpperCase());
+
+                                                org.json.JSONObject permissionsObjectForMqtt =
+                                                        extractPermissionsFromSolaceApplicationGetResponse(
+                                                        responseForMqtt, i, gatewayEnvironmentMap);
+
+                                                if (permissionsObjectForMqtt != null) {
+                                                    populateSolaceTopics(solaceTopicsObjectDTO,
+                                                            permissionsObjectForMqtt,
+                                                            SolaceConstants.MQTT_TRANSPORT_PROTOCOL_NAME.toUpperCase());
+                                                }
+                                            }
+                                            solaceDeployedEnvironmentsDTO.setSolaceTopicsObject(solaceTopicsObjectDTO);
+                                        }
+                                    }
+                                }
+                                solaceEnvironments.add(solaceDeployedEnvironmentsDTO);
+                            }
+                        }
+                    } catch (IOException e) {
+                        e.printStackTrace();
```

Comment:
Remove printStackTrace and add proper log with context. Do a full search and remove all please

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10842/commits/c41ba333ee522fd29e341c130570450703f787bf